DIPs/DIPxxxx_attributes.md

```diff
@@ -0,0 +1,144 @@
+
+A module declaration may be tagged with zero or more attribute groups, to apply to all symbols (bar templates which remain inferred with explicit tagging) declared within the module acting as the default.
+If any attribute groups are absent, then the value for that attribute group default to the corresponding value in `core.attribute.defaultAttributeSet`, which will have the values of the current defauls, but may be versioned in druntime as the end user wishes, of with command line switches (e.g. `-safe` or if Type_Info / Module Info generation is added as an attribute `-betterC`).
+
+As all the attributes are now symbols we can group the in an `AliasSeq` like fashion to apply them en masse, as is done in LDC for [`@fastmath`](https://github.com/ldc-developers/druntime/blob/ldc/src/ldc/attributes.d#L58).
+
+It is illegal to explicitly provide more than one attribute from any given attribute group as they are mutually exclusive. 
+Attributes applied explicity to any symbol override the module default attribute set.
+
+### Attributes & attribute like compiler behaviour encompassed in this DIP
+
+Encompassed:
+
+* pure
+* @nothrow
+* @nogc
+* @safe/@system/@trusted
+
+Optionally encompassed:
+
+* final
+* Type_Info / Module Info generation (other components of -betterC?)
+
+Not encompassed:
+
+* @disable
+* @property
+
+### Breaking changes / deprecation process
+
+Use of the current attributes that are not prefiex by an `@` such as `pure` and `nothrow`,
+and optionally other modifiers that are attribute like such as `final` will be changed to refer to the `core.attribute` symbols,
+and thus their use without the leading `@` will be deprecated. 
+
+No breaking changes are intended, although the introduction of the new enum symbols to be implicitly imported by `object.d`
+may break some code if the names chosen clash (unlikely).
+
+### Examples
+
+`module foo;` 
+
+will become implicitly 
+
+`@core.attribute.defaultAttributeSet module foo;` 
+
+with respect to attributes (`core.attribute` will be implicitly imported, by `object.d`), 
+if no attributes from `core.attribute` are attached.
+
+ Attribute groups may be selectivly added to the module declaration, so that:
+ 
+ `@nogc module foo;` 
+ 
+ means that all symbols in this module are implicity `@nogc` (with `nogc` referring to `core.attribute.GarbageCollectedness.nogc` via an alias in `core.attribute`),
+ but otherwise has all the same defaults as the default attribute set.
+ 
+ `@nogc @core.attribute.GarbageCollectedness.gc module foo;` 
+ shall be an error because there are two explicit mutually exclusive attributes.
+ Likewise 
+ ```
+ module foo;
+ @nogc @core.attribute.GarbageCollectedness.gc 
+ void bar() { };
+ ```
+ shall be an error for the same reasons.
+ 
+ ```
+ @nogc module foo;
+ 
+ // bar overrides the default @nogc'ness of the module and is @gc
+ @core.attribute.GarbageCollectedness.gc void bar() {auto a = new int;} 
```

Comment:
`@trusted:` is always a bad idea and should be deprecated. I don't think this DIP is orthogonal to issues with defaulting `@trusted`. In particular, by introducing attribute sets users can hide the `@trusted` attribute:
```d
@mySet module foo;
```
Here we allow trusted code in a module without being able to `grep` for it (something Walter often mentions for code audits). The DIP could avoid this by disallowing `@trusted` in an attribute set.